src/Microsoft.IdentityModel.Clients.ActiveDirectory/Internal/Instance/InstanceDiscovery.cs

```diff
@@ -187,11 +187,15 @@ private async Task DiscoverAsync(Uri authority, bool validateAuthority, RequestC
                 // All other unexpected exceptions will still bubble up, as always.
                 if (validateAuthority)
                 {
-                    // hard stop here
-                    throw new AdalException(
-                        (ex.ErrorCode == "invalid_instance")
-                            ? AdalError.AuthorityNotInValidList
-                            : AdalError.AuthorityValidationFailed, ex);
+                    if (ex.ErrorCode == ErrorCodes.InvalidInstance)
+                    {
+                        throw new AdalException(
+                            (ex.ErrorCode == "invalid_instance")
+                                ? AdalError.AuthorityNotInValidList
+                                : AdalError.AuthorityValidationFailed, ex);
+                    }
+
+                    throw new AdalException(ex.Response.StatusCode.ToString(), ex.InnerException.Message);                   
```

Comment:
You are already in a `catch` block where you caught the relevant exception. So instead of creating a new exception, just let the old one propagate further. There are 2 ways of doing it: 

`throw;` // preserves the original stack trace
`throw ex;` // resets the stack trace

It's not very important, but you should preserve the original stack trace.

Reply:
sounds good. thanks

---
In reply to: [356229220](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1682#discussion_r356229220) [](ancestors = 356229220)